modules/browser_object_navigation.py

```diff
@@ -0,0 +1,109 @@
+from pypom import Page
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class Navigation(Page):
+    """Page Object Model for nav buttons and AwesomeBar"""
+
+    URL_TEMPLATE = "about:blank"
+    BROWSER_MODES = {
+        "Bookmarks": "*",
+        "Tabs": "%",
+        "History": "^",
+        "Actions": ">",
+    }
+
+    _awesome_bar = (By.ID, "urlbar-input")
+    _xul_source_snippet = (
+        'xmlns:xul="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"'
+    )
+
+    # Tab-to-search
+    _tab_to_search_text_span = (
+        By.CLASS_NAME,
+        "urlbarView-dynamic-onboardTabToSearch-text-container",
+    )
+    _search_mode_span = (By.ID, "urlbar-search-mode-indicator-title")
+
+    # Overflow menu
+    _overflow_item = (
+        By.CSS_SELECTOR,
+        '[class="urlbarView-title urlbarView-overflowable"]',
+    )
+
+    # Search Mode One-Offs
+    _search_one_off_settings_button = (By.ID, "urlbar-anon-search-settings")
+
+    def ensure_chrome_context(self):
+        if self._xul_source_snippet not in self.driver.page_source:
+            self.driver.set_context(self.driver.CONTEXT_CHROME)
+
+    def resume_content_context(self):
+        if self._xul_source_snippet in self.driver.page_source:
+            self.driver.set_context(self.driver.CONTEXT_CONTENT)
```

Comment:
I am interested in why you chose to do the chrome context setting this way. Couldn't you just use the generator selenium provides for this? That way you couldn't have to manage switching back to the `content` context

Reply:
This was done before I started "fluentifying" my object model, I'll change it.

Reply:
Oh wait, no there was a reason—I will be returning an object (self) from inside these functions, and if I return that from within the Context Manager, Selenium gets amnesia about elements stored as class attributes and can no longer interact with them. I'll look at other ways to control the context here in future PRs.

Reply:
I guess what I am saying is to just enable chrome context when needed instead of switching to/from chrome. But this isn't a blocker, more so just curious on your thoughts.